test/test_base_metric.py

```diff
@@ -4,71 +4,170 @@
 import metric_learn
 import numpy as np
 from sklearn import clone
-from sklearn.utils.testing import set_random_state
+import sklearn
+from packaging import version
 from test.test_utils import ids_metric_learners, metric_learners, remove_y
+skversion = version.parse(sklearn.__version__)
+if skversion >= version.parse('0.22.0'):
+    from sklearn.utils._testing import set_random_state
+else:
+    from sklearn.utils.testing import set_random_state
 
 
 def remove_spaces(s):
   return re.sub(r'\s+', '', s)
 
 
+def sk_repr_kwargs(def_kwargs, nndef_kwargs):
```

Comment:
Here I thought it could be good to test the str repr for both the old (<=0.22) sklearn version and the newer ones, so I made a string representation that depends on the sklearn version